packages/desktop-client/src/components/ScrollProvider.tsx

```
@@ -1,61 +1,206 @@
-    (tolerance = 1) => scrollHeight - scrollY <= clientHeight + tolerance,
-    [clientHeight, scrollHeight, scrollY],
+  const hasScrolledToEnd = useCallback(
+    (direction: ScrollDirection, tolerance = 1) => {
+      switch (direction) {
+        case 'up':
+          const hasScrolledToTop = () => {
+            if (scrollY.current !== undefined) {
+              return scrollY.current <= tolerance;
+            }
+            return false;
+          };
```

Biome

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)